react-web/src/pages/auditing/reportUpload/ReportUpload.tsx

```diff
@@ -0,0 +1,317 @@
+  };
+
+  // eslint-disable-next-line
+  const [showError, setShowError] = useState("");
+
+  const form = useForm({
+    schema: reportUploadSchema,
+    mode: "onChange",
+  });
+
+  const { fields, append, remove } = useFieldArray({
+    control: form.control,
+    name: fieldArrayName,
+  });
+
+  const addNewDappScript = () => {
+    append(
+      {
+        scriptHash: "",
+        contactAddress: "",
+      },
+      { shouldFocus: true }
+    );
+  }
+
+  const formHandler = (formData: any) => {
+    const {subject, certificationLevel, name, logo, email, website, twitter, reportURL, summary, disclaimer, dAppScripts} = formData;
+    const formattedDappScripts: IScriptObject[] = [];
+    dAppScripts.forEach((script: any) => {
+      const {scriptHash, contactAddress, ...rest} = script
+      formattedDappScripts.push({
+        scriptHash: scriptHash,
+        contactAddress: contactAddress,
+        smartContractInfo: {
+          ...rest
+        }
+      })
+    });
+    const payload: OffChainMetadataSchema = {
+      subject: subject,
+      schemaVersion: 1,
+      certificationLevel: parseInt(certificationLevel),
+      certificateIssuer: {
+        name: name,
+        logo: logo,
+        social: {
+          contact: email,
+          link: website,
+          twitter: twitter || "",
+          github: website,
+          website: website
+        }
+      },
+      report: reportURL.replace(/\s+/g,"").split(','),
+      summary: summary,
+      disclaimer: disclaimer,
+      scripts: formattedDappScripts
+    }
+
+    fetchData.post("/auditor/reports", payload).then((response) => {
+      setShowError("")
+      setOpenModal(true)
+    }).catch((errorObj) => {debugger
+      let errorMsg = 'Something went wrong. Please try again.'
+        if (errorObj?.response?.data) {
+          errorMsg = errorObj.response.statusText + ' - ' + errorObj.response.data 
+        }
+        setShowError(errorMsg);
+        const timeout = setTimeout(() => { clearTimeout(timeout); setShowError("") }, 5000)
+        
+    })
+  };
+
+  const initializeFormState = () => {
+    const { twitter, website } = userDetails // TBD - subject, name, contact
+    let formData: any = { twitter, website }
+
+    form.reset(formData)
+  }
+
+  useEffect(() => {
+    initializeFormState()
+    // initializeFormState() is to not to be triggered on every re-render of the dep-array below but whenever the form or userDetails is updated alone
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [userDetails, form]);
+
+  // const [files, setFiles] = useState<any>();
+  // const uploadReport = () => {};
+
+  const certificationLevelOptions = [
+    {
+      label: "L0 Audit",
+      value: 0,
+    },
+    {
+      label: "L2 Certification",
+      value: 2,
+    },
+  ];
+
+
+  return (
+    <>
+      <h2>Upload an Audit Report</h2>
+      <div id="auditReportUploadContainer">
+        <Form form={form} onSubmit={formHandler}>
+          <Dropdown
```

Comment:
Not sure where it is, maybe in the Dropdown itself but can we get rid of the "x" that allows to clear the field?

Reply:
Already done